libs/hwServer/manufacturers/ledger/index.js

```diff
@@ -72,26 +71,28 @@ const getLedgerAccount = (index = 0) => {
 };
 
 /**
- * checkIfInsideLiskApp - function - Validate if after using the pin to unblock ledger device
+ * checkLiskAppStatus - function - Validate if after using the pin to unblock ledger device
  * the user is inside the LSK App, if not then will show the device as connected but not
  * able to get accounts from the device.
  * @param {object} param - Object with 2 elements, a transport and device.
  * @param {object} param.transporter - Object for handle the ledger device.
  * @param {object} param.device - Object with device information.
  */
-const checkIfInsideLiskApp = async ({ transporter, device }) => {
+// eslint-disable-next-line max-statements
+const checkLiskAppStatus = async ({ transporter, path }) => {
   let transport;
+  let status
   try {
-    transport = await transporter.open(device.path);
+    transport = await transporter.open(path);
     const liskLedger = new LiskApp(transport);
-    const ledgerAccount = getLedgerAccount();
-    const account = await liskLedger.getAddressAndPubKey(ledgerAccount.derivePath());
-    device.status = account ? DEVICE_STATUS.CONNECTED : DEVICE_STATUS.DISCONNECTED;
+    const ledgerAccount = getLedgerAccount(0);
+    const hasAccount = await liskLedger.getAddressAndPubKey(ledgerAccount.derivePath());
+    status = hasAccount ? DEVICE_STATUS.CONNECTED : DEVICE_STATUS.STAND_BY;
```

Comment:
We probably need to do something like this: 
`status = hasAccount.address ? DEVICE_STATUS.CONNECTED : DEVICE_STATUS.STAND_BY;`
or
`status = hasAccount.error_message === 'No errors' ? DEVICE_STATUS.CONNECTED : DEVICE_STATUS.STAND_BY;`

Reply:
but there is no error, (i think while the device path is not exist its return error nothing to do with firmware is connected or not) we keep the same logic as we had in past

Reply:
`await liskLedger.getAddressAndPubKey(ledgerAccount.derivePath());`

If inside lisk app it will return
```
{
  pubKey: '6cea247907980c412ec1b0f51957f7bd50391361a8ab2408b584f182b89e5f32',
  address: 'lsk3hm2t58ze3gs848hsbjca5dc2nc7uqedqdfehm',
  return_code: 36864,
  error_message: 'No errors'
}
```
If  ledger is open but **not** inside lisk app it will return:
```
{
  returnCode: 28161,
  errorMessage: 'Unknown Status Code: 28161',
  return_code: 28161,
  error_message: 'Unknown Status Code: 28161'
}
```

If ledger is disconnected it will throw (i think)

Reply:
So since `hasAccount` is an object in both these states: 
1.  Inside lisk app. 
2.  Not inside lisk app  

standby will never be true